src/main/java/org/prebid/server/util/BidderUtil.java

```diff
@@ -10,4 +19,32 @@ private BidderUtil() {
     public static boolean isValidPrice(BigDecimal price) {
         return price != null && price.compareTo(BigDecimal.ZERO) > 0;
     }
+
+    public static PriceFloorInfo resolvePriceFloor(Bid bid, BidRequest bidRequest) {
+        final String bidImpId = ObjectUtil.getIfNotNull(bid, Bid::getImpid);
+        if (StringUtils.isEmpty(bidImpId)) {
+            return null;
+        }
+
+        final List<Imp> imps = ObjectUtil.getIfNotNull(bidRequest, BidRequest::getImp);
+        if (CollectionUtils.isEmpty(imps)) {
+            return null;
+        }
+
+        final Imp correspondingImp = imps.stream()
+                .filter(imp -> Objects.equals(bidImpId, imp.getId()))
+                .findFirst()
+                .orElse(null);
+        if (correspondingImp == null) {
+            return null;
+        }
```

Comment:
Instead of using this if we can continue work with Optional from findFirst with .map method
```java
   return imps.stream()
                .filter(imp -> Objects.equals(bidImpId, imp.getId()))
                .findFirst()
                .map(BidderUtil::createFloorInfo)
                .orElse(null);
    }

    private static PriceFloorInfo createFloorInfo(Imp imp){
        final BigDecimal floor = imp.getBidfloor();
        final String currency = imp.getBidfloorcur();
        if (floor == null && currency == null) {
            return null;
        }

        return PriceFloorInfo.of(floor, currency);
    }
```

Reply:
Crazy cool! Like it!